File: classes01/cpp/rotate.cpp
#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;

int main(void)
{   
    // 原始图像
    Mat source = imread("../../data/images/sample.jpg");

    // 计算旋转变换的矩阵
    Mat M=getRotationMatrix2D(Point(source.cols/2, source.rows/2), 30,1.0);

    // 变换后的图像
    Mat dst;
    warpAffine(source,dst,M,Size(source.cols,source.rows));

    namedWindow("Original", WINDOW_AUTOSIZE);
    namedWindow("Rotation", WINDOW_AUTOSIZE);

    imshow("Original", source);
    imshow("Rotation", dst);

    waitKey(0);
    destroyAllWindows();

    

    return EXIT_SUCCESS;
}

File: classes01/python/rotate.py
import cv2

source=cv2.imread("../../data/images/sample.jpg")

# 获取图片的尺寸
dim=source.shape
M=cv2.getRotationMatrix2D((dim[1]/2, dim[0]/2),-30,1.0)
dst=cv2.warpAffine(source,M,(dim[1],dim[0]))

cv2.imshow("Original", source)
cv2.imshow("Rotate", dst)

cv2.waitKey(0)
cv2.destroyAllWindows()